explainability-service/src/main/java/org/kie/trustyai/service/endpoints/metrics/DisparateImpactRatioEndpoint.java

```diff
@@ -68,21 +71,25 @@ public String getMetricName() {
     public Response dir(@ValidBaseMetricRequest BaseMetricRequest request) throws DataframeCreateException {
 
         final Dataframe dataframe;
+        final Metadata metadata;
         try {
             dataframe = dataSource.get().getDataframe(request.getModelId());
+            metadata = dataSource.get().getMetadata(request.getModelId());
         } catch (DataframeCreateException e) {
             LOG.error("No data available for model " + request.getModelId() + ": " + e.getMessage(), e);
             return Response.serverError().status(Response.Status.BAD_REQUEST).entity("No data available").build();
         }
 
+        ReconciledMetricRequest reconciledMetricRequest = ReconciledMetricRequest.reconcile(request, metadata);
+
         final double dir;
         try {
-            dir = calculator.calculateDIR(dataframe, request);
+            dir = calculator.calculateDIR(dataframe, reconciledMetricRequest);
         } catch (MetricCalculationException e) {
             LOG.error("Error calculating metric for model " + request.getModelId() + ": " + e.getMessage(), e);
             return Response.serverError().status(Response.Status.BAD_REQUEST).entity("Error calculating metric").build();
         }
-        final String dirDefinition = calculator.getDIRDefinition(dir, request);
+        final String dirDefinition = calculator.getDIRDefinition(dir, reconciledMetricRequest);
 
         MetricThreshold thresholds;
         if (request.getThresholdDelta() == null) {
```

Comment:
I don't think it is a problem but there are still some usages of `request` instead of `reconciledMetricRequest`. I expect similar data are the same but it might be a bit error prone for the future so I suggest to rename `request` as `rawRequest`. Wdyt?